features/steps/steps.py

```diff
@@ -221,3 +234,46 @@ def step_impl(context, related, relating, other_entity, condition):
                         errors.append(instance_structure_error(inst, rel.RelatingObject))
 
     handle_errors(context, errors)
+
+
+@then('Each {entity} {condition} be {directness} contained in {other_entity}')
```

Comment:
This is not incorporating the `... or subtypes` postfix we discussed in a call. It's also not immediately relevant of course, but since the gherkin is a bit speculative anyway, maybe the purpose of this PR can be to investigate how general we can get.

Reply:
I am not 100% sure what you mean here. 

Is it the following? 

`@then('Each {entity} {condition} be {directness} contained in {other_entity} or its subtypes')`
Then in our particular case, it would be:
`@then'Each IfcAlignment must be directly contained in IfcSite or its subtypes'
So the check will pass if the IfcAlignment is contained either in the IfcSite or in any subtype (via inheritance) of IfcSite?

Good idea, I'll try to add it.

Reply:
I didn't specifically add `... or subtypes`, as it's already checking the inheritance via is_a method.